cea/analysis/lca/embodied.py

```diff
@@ -203,30 +220,56 @@ def calculate_contributions(df, year_to_calculate):
 
     # calculate the embodied energy/emissions due to construction
     total_column = 'saver'
+
     ## calculate how many years before the calculation year the building was built in
     df['delta_year'] = year_to_calculate - df['YEAR']
     ## if it was built more than 60 years before, the embodied energy/emissions have been "paid off" and are set to 0
     df['confirm'] = df.apply(lambda x: calc_if_existing(x['delta_year'], SERVICE_LIFE_OF_BUILDINGS), axis=1)
     ## if it was built less than 60 years before, the contribution from each building component is calculated
-    df[total_column] = ((df['GHG_WALL_kgCO2m2'] * (df['area_walls_ext_ag'] + df['area_walls_ext_bg']) +
-                         df['GHG_WIN_kgCO2m2'] * df['windows_ag'] +
-                         df['GHG_FLOOR_kgCO2m2'] * df['floor_area_ag'] +
-                         df['GHG_BASE_kgCO2m2'] * df['floor_area_bg'] +
-                         df['GHG_PART_kgCO2m2'] * (df['floor_area_ag']+df['floor_area_bg']) * CONVERSION_AREA_TO_FLOOR_AREA_RATIO +
-                         df['GHG_ROOF_kgCO2m2'] * df['footprint']) / SERVICE_LIFE_OF_BUILDINGS) * df['confirm']
 
-    df[total_column] += (((df['floor_area_ag'] + df['floor_area_bg']) * EMISSIONS_EMBODIED_TECHNICAL_SYSTEMS) / SERVICE_LIFE_OF_TECHNICAL_SYSTEMS) * df['confirm']
+    # New changes are essential i.e. (1) The LCA of replacement which consider the specific Reference service life
```

Comment:
Also: `ni=(((SERVICE_LIFE_OF_BUILDINGS/Service_Life_Component)-1)+1)`
The -1 and the +1 just cancel out, leaving: `ni=SERVICE_LIFE_OF_BUILDINGS/Service_Life_Component`
Is that the intended result?

Reply:
The equation has been corrected as below:
ni=(SERVICE_LIFE_OF_BUILDINGS/Service_Life_Component)-1

Reply:
Hmmm but that's just the replacements, right? Otherwise if `SERVICE_LIFE_OF_BUILDINGS == Service_Life_Component` then the embodied emissions for that component will be 0.

Reply:
If we just want to calculate the construction emissions + replacements, the original equation makes sense.
If we want to calculate construction and replacements separately, the second equation makes sense, but then we need an additional step before that to calculate the construction emissions.

Reply:
I double checked the standards and literature, and noticed that I misinterpreted the equation. To consider the replacement phase in LCA calculations, the original equation from DIN EN 15978 ((section 9.3.3) is proposed as:

ni=(SERVICE_LIFE_OF_BUILDINGS/Service_Life_Component)-1
 
However, all the emissions from production, replacement and end-of-life can be packed into one equation, the one with both ”+1” and "-1" that I wrote originally, which didn’t make sense. In fact, the point of adding the “+1”  was to avoid a value of zero for the components that have an identical service life with the building. Hence, with a bit of modification, the original nonsense equation can be considered as a reference, for calculating the "ni" value for both new and retrofitted buildings. I suggest rewriting the equation as: 

ni= np.max(((SERVICE_LIFE_OF_BUILDINGS/Service_Life_Component)-1),1)
 
Note from literature:
LCEI is composed of the environmental impacts (EI) associated with production of all building components (Module A1-A3), their replacement (module B4) and end of life (modules C3 and C4) as well as the environmental impacts related with building operation (B6). This is translated by the following equation: LCEI = b ∑ i=1 kproduction,i + kEoL, i )* ( ni + 1 ) + QF,a*kop.*RSP*ERA) [https://doi.org/10.1016/j.buildenv.2020.107143]. 
 
The last part of this equation (LCEI) is calculated separately in CEA in operation.py

I updated the equation in the latest commit.

Reply:
Hi Martin,

I see that I forgot to push the above changes into Github from my branch.

Reply:
Thanks for clarifying that!